pkg/ring/batch.go

```diff
@@ -118,15 +134,28 @@ func (b *batchTracker) record(sampleTrackers []*itemTracker, err error) {
 	// The use of atomic increments here guarantees only a single sendSamples
 	// goroutine will write to either channel.
 	for i := range sampleTrackers {
+		verifyIfShouldReturn := func() {
```

Comment:
`verify` to me suggests a function without side-effects; this one has three.

Reply:
yeah..
i agree that this logic is hard to understand.. lemme try to rewrite in another way.